### PR TITLE
Add SVM Workload

### PR DESCRIPTION
**Pull request by zhuoxiangchen**
SVM data generator is based on LogisticRegressionDataGenerator, and the code of SVM is from Spark.A preliminary test was conducted with following environment:
Cluster: 1 master, 3 workers, 3 executors and 46 cores for each executor.
Worker: CPU: Intel Xeon E5-2699 v3, Memory: 180G, Network: 1Gb
The test results:
Scale | Time | Data size
tiny | 18.4s | 7,7M
small | 23.9s | 763.5M
large | 4.73m | 74.5G
huge | 9.74m | 107.3G
gigantic | 17.02m | 146.0G
bigdata | 29.5m | 167.6G

### CODE REVIEW

File: sparkbench/ml/src/main/scala/com/intel/sparkbench/ml/SVMDataGenerator.scala [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+import org.apache.spark.{SparkConf, SparkContext}
+import org.apache.spark.annotation.{DeveloperApi, Since}
+import org.apache.spark.mllib.linalg.Vectors
+import org.apache.spark.mllib.regression.LabeledPoint
+import org.apache.spark.rdd.RDD
+
+/**
+ * :: DeveloperApi ::
+ * Generate test data for LogisticRegression. This class chooses positive labels
```
Comment (jtengyp) [inline sparkbench/ml/src/main/scala/com/intel/sparkbench/ml/SVMDataGenerator.scala]: The comment is not changed.

File: sparkbench/ml/src/main/scala/com/intel/sparkbench/ml/SVMDataGenerator.scala [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+ * with probability `probOne` and scales features for positive examples by `eps`.
+ */
+object SVMDataGenerator {
+
+  /**
+   * Generate an RDD containing test data for LogisticRegression.
```
Comment (jtengyp) [inline sparkbench/ml/src/main/scala/com/intel/sparkbench/ml/SVMDataGenerator.scala]: The comment is not changed.

File: sparkbench/ml/src/main/scala/com/intel/sparkbench/ml/SVMDataGenerator.scala [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+   *
+   * @param sc SparkContext to use for creating the RDD.
+   * @param nexamples Number of examples that will be contained in the RDD.
+   * @param nfeatures Number of features to generate for each example.
+   * @param eps Epsilon factor by which positive examples are scaled.
```
Comment (jtengyp) [inline sparkbench/ml/src/main/scala/com/intel/sparkbench/ml/SVMDataGenerator.scala]: The parameter eps is not called in this method.

File: sparkbench/ml/src/main/scala/com/intel/sparkbench/ml/SVMDataGenerator.scala [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+   * @param nparts Number of partitions of the generated RDD. Default value is 2.
+   * @param probOne Probability that a label is 1 (and not 0). Default value is 0.5.
+   */
+  def generateSVMRDD(
+    sc: SparkContext,
+    nexamples: Int,
+    nfeatures: Int,
+    eps: Double,
+    nparts: Int = 2,
+    probOne: Double = 0.5): RDD[LabeledPoint] = {
```
Comment (jtengyp) [inline sparkbench/ml/src/main/scala/com/intel/sparkbench/ml/SVMDataGenerator.scala]: The parameter probOne is not called in this function.

**Comment by jtengyp** [comment]
bin/run_all.sh is not taken into account.

File: conf/hibench.conf [hunk 5]
```diff
@@ -1,12 +1,11 @@
 # Data scale profile. Available value is tiny, small, large, huge, gigantic and bigdata.
 # The definition of these profiles can be found in the workload's conf file i.e. conf/workloads/micro/wordcount.conf
-hibench.scale.profile                 tiny
-
+hibench.scale.profile                large
```
Comment (VinceShieh) [inline conf/hibench.conf]: set to tiny

File: sparkbench/ml/src/main/scala/com/intel/sparkbench/ml/SVMWithSGDExample.scala [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    val metrics = new BinaryClassificationMetrics(scoreAndLabels)
+    val auROC = metrics.areaUnderROC()
+
+    println("Area under ROC = " + auROC)
+
+    // Save and load model
+    // model.save(sc, "target/tmp/scalaSVMWithSGDModel")
+    //val sameModel = SVMModel.load(sc, "target/tmp/scalaSVMWithSGDModel")
+    // $example off$
+
```
Comment (VinceShieh) [inline sparkbench/ml/src/main/scala/com/intel/sparkbench/ml/SVMWithSGDExample.scala]: remove all the inference, only keep the training stage

File: conf/hibench.conf [hunk 7]
```diff
@@ -1,12 +1,11 @@
...
 # The definition of these profiles can be found in the workload's conf file i.e. conf/workloads/micro/wordcount.conf
-hibench.scale.profile                 tiny
-
+hibench.scale.profile                large 
 # Mapper number in hadoop, partition number in Spark
 hibench.default.map.parallelism         8
 
 # Reducer nubmer in hadoop, shuffle partition number in Spark
-hibench.default.shuffle.parallelism     8
+hibench.default.shuffle.parallelism    92
```
Comment (VinceShieh) [inline conf/hibench.conf]: set to default 8

**Comment by VinceShieh** [comment]
put on hold this PR, please do not merge for now

**Pull request merged**